### ISSUE TITLE
S1144 - false positive "remove unused private ctor" in C# 9

### ISSUE THREAD

**Issue by xagon** [SonarSource/sonar-dotnet#5162]
### Description

Analyzer rule S1144. The new C# introduced shorty syntax fot ctor using `new()` syntax.

![image](https://user-images.githubusercontent.com/20299573/145251748-f3b0688c-9882-4f4a-a7c6-d7491f1cc7bc.png)

### Repro steps

```
namespace abc
{
    public sealed class Wildcard
    {
        public static readonly Wildcard Asterisk = new(0, "*", ".*");
        public static readonly Wildcard QuestionMark = new(1, "?", ".");

        private Wildcard(int id, string wildcard, string regexEquivalent)
        {
            RegexEquivalent = regexEquivalent;
        }

        public string RegexEquivalent { get; }
    }
}
```

### Expected behavior

No analyzer warning should be shown. Valid case in c# 9.

### Actual behavior

Analyzer warning.

### Known workarounds

n/a

### Related information

* Visual Studio version => 2022
* MSBuild / dotnet version => .net 6
* Operating System => Windows 10

**Comment by andrei-epure-sonarsource** [comment]
hello @xagon  - what version of the analyzer are you using? (or are you on SonarCloud?)

**Comment by xagon** [comment]
hi @andrei-epure-sonarsource ,
`8.9.3.48735 SonarVersion`
Would this be sufficient? Do you need more insights here?

**Comment by costin-zaharia-sonarsource** [comment]
Hi @xagon, thanks for your response.

This issue was solved in https://github.com/SonarSource/sonar-dotnet/releases/tag/8.24.0.32949 and an update to this version or a later one should fix the issue for you. 

This can be achieved by [updating the SonarQube to the latest version](https://www.sonarqube.org/downloads/).

**Comment by xagon** [comment]
Thanks! 
That's what we needed!
